bionic/core/task_execution.py

```diff
@@ -313,16 +278,21 @@ class EntryRequirement:
     """
     Represents a requirement from one entry to another.
 
-    A requirement indicates that one entry (`src_entry`) can't make any further progress
-    until another entry (`dst_entry`) has reached a certain level of progress (`level`).
+    A requirement indicates that we need a particular entry (``dst_entry``) to reach a
+    certain level of progress (``level``). This could be because it's blocking some
+    other entry (``src_entry``), or for some external or a priori reason.
 
-    Note: `src_entry` can also be `None`, indicating some sort of external or a priori
-    requirement. On the other hand, `dst_entry` cannot be `None`.
+    An entry can be marked ``is_transient``, meaning it can safely be discarded once
+    ``dst_entry`` has reached the target ``level``. Otherwise, it also be discarded if
+    its ``src_entry`` has reached the (final) ``CACHED`` level. Once a requirement is
+    discarded, it's permissible for ``dst_entry`` to revert to a previous level. If a
+    requirement is not transient and has no source, it is assumed to be permanent.
     """
 
-    src_entry = attr.ib()
-    dst_entry = attr.ib()
-    level = attr.ib()
+    dst_entry: TaskRunnerEntry = attr.ib()
+    level: "EntryLevel" = attr.ib()
```

Comment:
If this indicates the requirements of `dst_entry`, what about adding a `dst_` prefix?

Reply:
I think this could make sense. It might be better to do it in a separate PR though. Let's discuss it in our meeting today.